src/linker/implementation.py

```diff
@@ -48,6 +55,19 @@ def validate(self) -> List[Optional[str]]:
     # Helper methods #
     ##################
 
+    def _format_config(self, config: Optional[Dict[str, Any]]) -> Optional[Dict[str, str]]:
+        return self._stringify_keys_values(config) if config else None
+
+    def _stringify_keys_values(self, config: Any) -> Any:
```

Comment:
Just for fun, here is something you could do for the return type:

```suggestion
    StringValuedDictionary = Dict[str, Union[str, 'StringValuedDictionary']]

    def _stringify_keys_values(self, config: Any) -> StringValuedDictionary:
```

Reply:
Hah, I'm pretty torn on creating objects just to hold typing. But it is fun so I'll do it